File: features/snippets/external_storage/s3_setup/s3_external_storage_setup.go
package s3setup

import (
	"log"

	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/converter"
	"go.temporal.io/sdk/worker"
)

func SetupExternalStorage(driver converter.StorageDriver) {
	// @@@SNIPSTART go-s3-external-storage-setup
	c, err := client.Dial(client.Options{
		HostPort: "localhost:7233",
		ExternalStorage: converter.ExternalStorage{
			Drivers: []converter.StorageDriver{driver},
		},
	})
	if err != nil {
		log.Fatalf("connect to Temporal: %v", err)
	}
	defer c.Close()

	w := worker.New(c, "my-task-queue", worker.Options{})
	// @@@SNIPEND
	_ = w
}

File: features/snippets/external_storage/s3_setup/s3_external_storage_setup.java
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowClientOptions;
import io.temporal.payload.storage.ExternalStorage;
import io.temporal.payload.storage.StorageDriver;
import io.temporal.serviceclient.WorkflowServiceStubs;
import io.temporal.worker.Worker;
import io.temporal.worker.WorkerFactory;

class S3ExternalStorageSetupSnippet {
  static void configureClientAndWorker(StorageDriver driver) {
    // @@@SNIPSTART java-s3-external-storage-setup
    ExternalStorage externalStorage = ExternalStorage.newBuilder().setDriver(driver).build();

    WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
    WorkflowClient client =
        WorkflowClient.newInstance(
            service,
            WorkflowClientOptions.newBuilder().setExternalStorage(externalStorage).build());
    WorkerFactory factory = WorkerFactory.newInstance(client);
    Worker worker = factory.newWorker("my-task-queue");
    // @@@SNIPEND
    factory.shutdown();
    service.shutdown();
  }
}
